src/index.ts

```diff
@@ -149,7 +149,15 @@ export function register (options: Options = {}): () => Register {
   const shouldCache = !!(options.cache == null ? DEFAULTS.cache : options.cache)
   const fast = !!(options.fast == null ? DEFAULTS.fast : options.fast)
   const project = options.project || DEFAULTS.project
-  const cacheDirectory = options.cacheDirectory || DEFAULTS.cacheDirectory || join(tmpdir(), 'ts-node')
+
+  function getTmpDir(): string {
```

Comment:
The reason for not using the username was mostly because there is a method to get the homedir provided by node, but not one for username. So I thought this would be the easiest way to get good behavior across platforms. Alternatively to find the username we could look through env variables, or somehow use [this libuv code](http://docs.libuv.org/en/v1.x/misc.html#c.uv_os_get_passwd) (not sure how easy the ffi would be for this...); but neither of these are things I feel like I could do well. I could attempt this if it is desirable though.

Additionally, I think that homedir should be a reasonable option. I don't think that shared homedir's will be a common environment, and even if there are some setups like that this will not be a regression.

I opted not to use the cwd because I thought it would be desirable to share the cache between projects run by the same user. Additionally, two users could run code from the same directory and have conflicting caches. The only case I can think of where this would be superior would be if the user's homedir is empty string, in which case using the cwd would probably be better. Could fall-back to using the cwd if the users homedir is empty string? At the cost of a little more complexity this should provide better behavior.

I will update the code per your comments.

Thanks,
David